streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -450,22 +450,6 @@ public KStream<K, V> peek(final ForeachAction<? super K, ? super V> action,
             builder);
     }
 
-    @Deprecated
-    @Override
-    @SuppressWarnings("unchecked")
-    public KStream<K, V>[] branch(final Predicate<? super K, ? super V>... predicates) {
-        return doBranch(NamedInternal.empty(), predicates);
-    }
-
-    @Deprecated
-    @Override
-    @SuppressWarnings("unchecked")
-    public KStream<K, V>[] branch(final Named named,
-                                  final Predicate<? super K, ? super V>... predicates) {
-        Objects.requireNonNull(named, "named can't be null");
-        return doBranch(new NamedInternal(named), predicates);
-    }
-
     @SuppressWarnings({"unchecked", "rawtypes"})
     private KStream<K, V>[] doBranch(final NamedInternal named,
```

Comment:
@frankvicky -- this method in unused now -- can you do a follow up PR to remove it (using the same Jira).

Reply:
Cool. Thx. Please share a link to the PR here so I don't miss it :)

Reply:
Yeah, sure.
Here is the link:
https://github.com/apache/kafka/pull/16981